atlasdb-workload-server/src/test/java/com/palantir/atlasdb/workload/runner/AntithesisWorkflowValidatorRunnerTest.java

```diff
@@ -86,31 +90,47 @@ public void runValidatesAllInvariantsIgnoringExceptions() {
 
     @Test
     public void runExecutesMultipleWorkflowsAndWaitsForAllToFinishBeforeInvokingInvariantReporter() {
-        Semaphore semaphore = new Semaphore(0);
+        CountDownLatch slowWorkflowCanMakeProgress = new CountDownLatch(1);
+        AtomicBoolean slowWorkflowIsDone = new AtomicBoolean(false);
+
         Workflow slowWorkflow = mock(Workflow.class);
         when(slowWorkflow.run()).thenAnswer(_input -> {
             try {
-                assertThat(semaphore.tryAcquire(5, TimeUnit.SECONDS))
-                        .as("No permits should have been released yet as the invariant reporter should have not ran")
-                        .isFalse();
+                slowWorkflowCanMakeProgress.await();
+                slowWorkflowIsDone.set(true);
+                return workflowHistory;
             } catch (InterruptedException e) {
+                Thread.currentThread().interrupt();
                 throw new RuntimeException(e);
             }
-            return workflowHistory;
         });
 
-        WorkflowValidator<Workflow> slowWorkflowValidator = WorkflowValidator.of(slowWorkflow, invariantReporterOne);
-
-        doAnswer(_input -> {
-                    semaphore.release();
+        doAnswer(_invocation -> {
+                    assertThat(slowWorkflowIsDone.get())
+                            .as("the invariant reporter ran, even though the slow workflow was not done yet")
+                            .isTrue();
                     return null;
                 })
                 .when(invariantReporterOne)
                 .report(any());
 
-        new AntithesisWorkflowValidatorRunner(EXECUTOR_SERVICE).run(slowWorkflowValidator, workflowValidator);
+        ExecutorService backgroundExecutor = PTExecutors.newSingleThreadExecutor();
+        try {
+            Future<Void> validation = backgroundExecutor.submit(() -> {
+                WorkflowValidator<Workflow> slowWorkflowValidator =
+                        WorkflowValidator.of(slowWorkflow, invariantReporterOne);
+                new AntithesisWorkflowValidatorRunner(EXECUTOR_SERVICE).run(slowWorkflowValidator, workflowValidator);
+                return null;
+            });
+
+            assertThat(validation).as("the slow workflow was not actually slow").isNotDone();
+
+            slowWorkflowCanMakeProgress.countDown();
```

Comment:
sure, we can do it this way. The only reason I avoided this approach, was b/c it doesn't necessarily catch the race always; `slowWorkflowCanMakeProgress` could have a countdown before actually executing, thus would immediately go through